spec/support/vcr_config.rb

```diff
@@ -47,4 +47,13 @@ def test_docker_uri
     uri2.query = nil
     uri1.to_s.gsub(/\/\d{3}/,"") == uri2.to_s.gsub(/\/\d{3}/,"")
   end
+
+  config.register_request_matcher :s3_image_matcher do |request1, request2|
+    uri1 = URI(request1.uri)
+    uri2 = URI(request2.uri)
+
+    uri_regex = /^\/artworks\/(original|thumb)\/_hey-hey-[0-9]+__[A-z0-9]+\.png/
+  
+    uri_regex.match?(uri1.path) && uri_regex.match?(uri2.path)
```

Comment:
I'm still not convinced that this is correct.  suppose both `request1` and `request2` both return the same exact image, with the path `/artworks/original/ ... png`.  in that case, this condition will then return `true`.

but what we actually want to test for is that _both_ of the images are returned, both `/artworks/original/...` and `/artworks/thumb/...`. 

I think what you want to test for is that one of the requests is `/original/ `and the other is `/thumb/`.  I'm not sure if the request order is guaranteed, so you would probably need to do something like:
```
uri_regex_original = /^\/artworks\/original\/_hey-hey-[0-9]+__[A-z0-9]+\.png/
uri_regex_thumb = /^\/artworks\/thumb\/_hey-hey-[0-9]+__[A-z0-9]+\.png/

original_received = uri_regex_original.match?(uri1.path) && uri_regex_original.match?(uri2.path)
thumb_received = uri_regex_thumb.match?(uri1.path) && uri_regex_thumb.match?(uri2.path)

original_received && thumb_received
```
this would ensure that both the `/original/` and `/thumb/` images were in the request

Reply:
actually, nvm I'm an idiot, I see what you're saying.  the urls in the cassette, one of them being /original/ and one of them being /thumb/ are compared against this matcher.  so both are in fact checked.